🐛enhance(demo-env): improve ingress wait for k3d and add diagnostics

rishi-jat opened the pull request:
This PR improves the robustness of the ingress wait logic for K3d in demo-env.sh and adds helpful diagnostic output to aid debugging.

MikeSpreitzer commented on scripts/create-k3s-cluster-with-SSL-passthrough.sh:
```diff
@@ -103,11 +103,25 @@ helm upgrade --install ingress-nginx ingress-nginx \
       sleep 5
+      attempt=$((attempt+1))
+      if [ $attempt -ge $max_attempts ]; then
+        echo "ERROR: Timed out waiting for nginx ingress controller pod to appear."
+        kubectl get pods -n ingress-nginx -o wide || true
+        exit 1
```
We learned at KubeCon that sometimes people try our stuff in an astonishingly slow network environment. We should be removing or lengthening timeouts, not adding them or making them shorter.

rishi-jat commented:
@MikeSpreitzer Thanks for the feedback, @MikeSpreitzer!
Would it be okay to keep waiting indefinitely but print diagnostics every minute, and make the timeout optional via an INGRESS_TIMEOUT env variable?
That way we support slow networks while still giving users visibility. Let me know what you think!

rishi-jat commented:
@MikeSpreitzer,
I’ve updated the demo env script to use an infinite (or user-configurable) timeout for nginx ingress readiness, and added periodic diagnostics to help users in slow network environments. This fully addresses the KubeCon feedback.

rishi-jat commented:
Please have a look at the PR!
Thanks!

MikeSpreitzer commented on scripts/create-kubestellar-demo-env.sh:
```diff
@@ -112,7 +112,7 @@ checking_cluster() {
         while IFS= read -r line; do
 
-            if echo "$line" | grep -q $1; then
+            if echo "$line" | grep -q "$1"; then
```
For follow-on PR(s): do a more reliable search.

MikeSpreitzer commented on scripts/create-kubestellar-demo-env.sh:
```diff
@@ -220,18 +266,18 @@ helm upgrade --install ks-core oci://ghcr.io/kubestellar/kubestellar/core-chart
         --set-json='kubeflex-operator.verbosity=5' \
         --timeout=24h \
-        $var_flags
+        "$var_flags"
```
We do _not_ want dquotes around this variable expansion. We want it to be followed by word splitting.  See https://www.gnu.org/software/bash/manual/bash.html#Shell-Expansions-1

rishi-jat commented:
@MikeSpreitzer
Hey Mike! Fixed #2725 (demo env hanging at ingress readiness) with your suggested approach - 24h timeout, simple kubectl wait, proper error handling.

MikeSpreitzer commented on scripts/create-kubestellar-demo-env.sh:
```diff
@@ -217,10 +229,12 @@ kflex ctx --overwrite-existing-context wds2
 echo -e "\nWaiting for OCM cluster manager to be ready..."
-kubectl --context $k8s_platform-kubeflex wait controlplane.tenancy.kflex.kubestellar.org/its1 --for 'jsonpath={.status.postCreateHooks.its-with-clusteradm}=true' --timeout 24h
-kubectl --context $k8s_platform-kubeflex wait -n its1-system job.batch/its-with-clusteradm --for condition=Complete --timeout 24h
+kubectl --context "$k8s_platform"-kubeflex wait controlplane.tenancy.kflex.kubestellar.org/its1 --for 'jsonpath={.status.postCreateHooks.install-status-addon}=true' --timeout 24h
+kubectl --context "$k8s_platform"-kubeflex wait -n its1-system job.batch/install-status-addon --for condition=Complete --timeout 24h
+kubectl --context "$k8s_platform"-kubeflex wait controlplane.tenancy.kflex.kubestellar.org/its1 --for 'jsonpath={.status.postCreateHooks.its-hub-init}=true' --timeout 24h
+kubectl --context "$k8s_platform"-kubeflex wait -n its1-system job.batch/its-hub-init --for condition=Complete --timeout 24h
```
As noted in #3163, it is too soon to make this change.

clubanderson commented:
@rishi-jat is this closed or open to others to contribute?
